File: xiaohashu-distributed-id-generator/xiaohashu-distributed-id-generator-biz/src/main/java/com/tongji/xiaohashu/distributed/id/generator/biz/constant/RedisKeyConstants.java
package com.tongji.xiaohashu.distributed.id.generator.biz.constant;

/** 
* @time 2025/3/27 10:19
* @author tongji
* @description 
*/
public class RedisKeyConstants {
    /**
     * 生成 id 前缀
     */
    public static final String LEAF_REDIS_ID_KEY_PREFIX = "leaf:";

    /**
     * 分布式锁前缀
     */
    public static final String LOCK_LEAF_KEY_PREFIX = "lock:";
}

File: xiaohashu-user/xiaohashu-user-biz/src/main/java/com/tongji/xiaohashu/user/biz/constant/RedisKeyConstants.kt
package com.tongji.xiaohashu.user.biz.constant

object RedisKeyConstants {
    /**
     * 小哈书全局 ID 生成器 KEY
     */
    const val XIAOHASHU_ID_GENERATOR_KEY: String = "xiaohashu.id.generator"

    /**
     * 用户角色数据 KEY 前缀
     */
    private const val USER_ROLES_KEY_PREFIX = "user:roles:"

    /**
     * 用户对应的角色集合 KEY
     */
    @JvmStatic
    fun buildUserRoleKey(userId: Long) = "$USER_ROLES_KEY_PREFIX$userId"

    /**
     * 角色对应的权限集合 KEY 前缀
     */
    private const val ROLE_PERMISSIONS_KEY_PREFIX = "role:permissions:"


    /**
     * 构建角色对应的权限集合 KEY
     */
    @JvmStatic
    fun buildRolePermissionsKey(roleKey: String) = "$ROLE_PERMISSIONS_KEY_PREFIX$roleKey"

    /**
     * 用户信息数据 KEY 前缀
     */
    private const val USER_INFO_KEY_PREFIX: String = "user:info:"

    /**
     * 构建角色对应的权限集合 KEY
     */
    @JvmStatic
    fun buildUserInfoKey(userId: Long) = "$USER_INFO_KEY_PREFIX$userId"
}